### PR TITLE
Add missing  usdc on solana

### PR DESCRIPTION
**Pull request by ruijialin-avalabs**
## Description

**Ticket: [CP-]**

Please provide:

- A summary of the changes
- Motivation and context for this change
- Any dependencies introduced
- (If breaking) migration steps and instructions

- show usdc on solana to meld onramp flow when there is no usdc on solana balance available

## Screenshots/Videos

Include relevant screenshots or screen recordings of iOS and Android.

## Testing

**Dev Testing (if applicable)**

- Provide steps to test the happy path of your feature
- Provide steps to test edge cases and error states
- Trigger a build on bitrise and reference it here
- Move the ticket into the "Testing" column on Jira

- onRamp
    - go to account with SOL and USDC SOL
    - verify SOL and USDC SOL are in the select a token list
    - go to account with no SOL and USDC SOL
    - verify SOL and USDC SOL are in the select a token list

- offramp:
    - go to account with SOL and USDC SOL
    - verify SOL and USDC SOL are in the select a token list
    - go to account with no SOL and USDC SOL
    - verify SOL and USDC SOL are not in the select a token list
  
**QA Testing (if applicable)**

- Provide instructions for QA to test this feature thoroughly
- State expected behavior / acceptance criteria

## Checklist

Please check all that apply (if applicable)

- [X] I have performed a self-review of my code
- [X] I have verified the code works
- [X] I have included screenshots / videos of android and ios
- [X] I have added testing steps
- [ ] I have added/updated necessary unit tests
- [ ] I have updated the documentation

### CODE REVIEW

File: packages/core-mobile/app/new/features/meld/components/TokenList.tsx [hunk 1]
```diff
@@ -37,27 +42,80 @@ export const TokenList = ({
...
   })
 
+  const filteredUsdcSolanaToken = useMemo(() => {
+    return filteredTokenList.find(
+      tk =>
+        // @ts-expect-error: there is no chainId for SPL tokens
+        tk.address === USDC_SOLANA_TOKEN_ID &&
+        tk.networkChainId === ChainId.SOLANA_MAINNET_ID
+    )
+  }, [filteredTokenList])
```
Comment (atn4z7) [inline packages/core-mobile/app/new/features/meld/components/TokenList.tsx:82]: isn't this the same as usdcSolanaToken above?

File: packages/core-mobile/app/new/features/meld/components/TokenList.tsx [hunk 2]
```diff
@@ -37,27 +42,80 @@ export const TokenList = ({
...
-    useSupportedCryptoCurrencies()
+  const { tokenIndex, splTokenIndex, setTokenIndex, setSplTokenIndex } =
+    useTokenIndex()
+  const {
+    supportedErc20AndNativeCryptoCurrencies,
+    supportedSplCryptoCurrencies,
+    setSupportedCryptoCurrencies,
+    setSupportedSplCryptoCurrencies
+  } = useSupportedCryptoCurrencies()
+  const isSolanaSwapBlocked = useSelector(selectIsSolanaSwapBlocked)
```
Comment (atn4z7) [inline packages/core-mobile/app/new/features/meld/components/TokenList.tsx:53]: let's use solana support flag instead of swap

File: packages/core-mobile/app/new/features/meld/components/TokenList.tsx (hunk 1 above)
Comment (ruijialin-avalabs) [reply packages/core-mobile/app/new/features/meld/components/TokenList.tsx:82]: one is checking chainId and one is checking networkChainId

Comment (atn4z7) [reply packages/core-mobile/app/new/features/meld/components/TokenList.tsx:82]: but what is the difference? can't we just use one of them?
